fetcher.go

```diff
@@ -137,7 +153,43 @@ func (p *pool) fetchResource(ctx context.Context, from string, resource string,
 				fetchDurationPerCarPerPeerSuccessMetric.WithLabelValues(cacheStatus).Observe(float64(response_success_end.Sub(start).Milliseconds()))
 			}
 
-			updateSuccessServerTimingMetrics(respHeader.Values(servertiming.HeaderKey), resourceType, isCacheHit, durationMs, ttfbMs, received)
+			latencyMillis, speedBytesPerMs := updateSuccessServerTimingMetrics(respHeader.Values(servertiming.HeaderKey), resourceType, isCacheHit, durationMs, ttfbMs, received)
+
+			if latencyMillis > 0 && speedBytesPerMs > 0 {
+				p.lk.Lock()
+				if _, ok := p.uniquePerf[from]; !ok {
+					node := p.nodesPerf[from]
+					node.nSuccess++
+					node.latencyDigest.Add(latencyMillis, 1)
+					node.throughputDigest.Add(speedBytesPerMs, 1)
+					pct := (node.nSuccess * 100) / node.nRequests
+					if pct >= 70 && node.nRequests >= 30 {
+						p.uniquePerf[from] = struct{}{}
+						fetchPeerLatencyDistributionMetric.WithLabelValues("P25").Observe(node.latencyDigest.Quantile(0.25))
+						fetchPeerLatencyDistributionMetric.WithLabelValues("P50").Observe(node.latencyDigest.Quantile(0.50))
+						fetchPeerLatencyDistributionMetric.WithLabelValues("P75").Observe(node.latencyDigest.Quantile(0.75))
+						fetchPeerLatencyDistributionMetric.WithLabelValues("P90").Observe(node.latencyDigest.Quantile(0.90))
+						fetchPeerLatencyDistributionMetric.WithLabelValues("P95").Observe(node.latencyDigest.Quantile(0.95))
+						fetchPeerLatencyDistributionMetric.WithLabelValues("P99").Observe(node.latencyDigest.Quantile(0.99))
+
+						fetchPeerSpeedDistributionMetric.WithLabelValues("P25").Observe(node.throughputDigest.Quantile(0.25))
+						fetchPeerSpeedDistributionMetric.WithLabelValues("P50").Observe(node.throughputDigest.Quantile(0.50))
+						fetchPeerSpeedDistributionMetric.WithLabelValues("P75").Observe(node.throughputDigest.Quantile(0.75))
+						fetchPeerSpeedDistributionMetric.WithLabelValues("P90").Observe(node.throughputDigest.Quantile(0.90))
+						fetchPeerSpeedDistributionMetric.WithLabelValues("P95").Observe(node.throughputDigest.Quantile(0.95))
+						fetchPeerSpeedDistributionMetric.WithLabelValues("P99").Observe(node.throughputDigest.Quantile(0.99))
+						// latency < 200ms
+						if node.latencyDigest.Quantile(0.90) < 200 {
+							fetchPeerP90GoodLatencyCountMetric.Add(1)
+						}
+						// > 1MB/s
```

Comment:
i think as we've seen from current graphs we don't trust this one enough to feel confident in using this 'good speed count number' for anything yet